cfme/tests/infrastructure/test_infra_object_tag_visibility.py

```diff
@@ -41,14 +41,20 @@ def testing_vis_object(request, a_provider, appliance):
     if not test_items:
         pytest.skip('No content found for test!')
 
-    if collection_name == 'templates':
+    if collection_name in ['templates', 'vms']:
         item_type = a_provider.data['provisioning']['catalog_item_type'].lower()
         for test_item_value in test_items:
             if test_item_value.vendor == item_type:
                 return param_class(name=test_item_value.name, provider=a_provider)
     elif collection_name != 'providers':
-        return param_class(name=test_items[0].name, provider=a_provider)
-    return a_provider
+        try:
+            return param_class(name=test_items[0].name, provider=a_provider)
```

Comment:
As this moves forward ALL objects that have Collections, will have to be instantiated via the Collection and they will not take the appliance object, the collection does, but the instance object doesn't. This is a warning for the future.